records_mover/records/records_format.py

```diff
@@ -73,14 +73,14 @@ def __eq__(self, other: object) -> bool:
 
     def alter_hints(self, new_hints: UntypedRecordsHints) ->\
             'DelimitedRecordsFormat':
-        input_hints = dict(self.hints)  # make copy
+        input_hints = dict(self.custom_hints)  # make copy
         input_hints.update(new_hints)
         return DelimitedRecordsFormat(variant=self.variant,
                                       hints=input_hints)  # type: ignore
 
     def alter_variant(self, variant: str) -> 'DelimitedRecordsFormat':
         return DelimitedRecordsFormat(variant=variant,
-                                      hints=self.hints)  # type: ignore
+                                      hints=self.custom_hints)
```

Comment:
To make this work, we need to maintain a history of which hints were ever provided specifically.  Before, this history was being lost.  Indeed, this resulted in some unfortunate behavior.  For instance:

```sh
mvrec table2file --target.dialect csv --target.no_compression  ...
```

resulted in no compression, whereas:

```sh
mvrec table2file --target.no_compression --target.dialect csv ...
```

resulted in GZIP compression.